packages/core/src/services/app/user/index.ts

```diff
@@ -361,40 +361,22 @@ export default class UserService {
             };
         }
 
-        let count = 0;
-        let rows: AppNotification[] = [];
-
-        if (unreadOnly !== undefined) {
-            const notifications = await models.appNotification.findAndCountAll({
-                where: {
-                    ...where,
-                    read: !unreadOnly
-                },
-                offset,
-                limit,
-                order: [['createdAt', 'DESC']]
-            });
-
-            count = notifications.count;
-            rows = notifications.rows as AppNotification[];
-        } else {
-            count = await models.appNotification.count({
-                where: {
-                    ...where,
-                    read: false
-                }
-            });
-            rows = await models.appNotification.findAll({
-                where,
-                offset,
-                limit,
-                order: [['createdAt', 'DESC']]
-            });
-        }
+        const notifications: {
+            rows: AppNotification[];
+            count: number;
+        } = await models.appNotification.findAndCountAll({
+            where: {
+                ...where,
+                ...(unreadOnly !== undefined ? { read: !unreadOnly } : {})
```

Comment:
I suggest you add this on the lines above, doing a validation like it is for `if (isWebApp !== undefined)` and `if (isWallet !== undefined)`, leaving this much clean with a simple `where`

Reply:
much better!